plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/OptInDialog.java

```diff
@@ -26,13 +28,28 @@ public OptInDialog(Shell parentShell) {
    */
   @Override
   protected Point getInitialLocation(Point initialSize) {
-    Rectangle parentBounds = getParentShell().getBounds();
-    Rectangle parentClientArea = getParentShell().getClientArea();
+    Shell targetShell = null;
+    try {
+      // Prefer showing at the top-right corner of the currently active workbench, but
+      // if we can't get a workbench for any reason, fall back to the parent shell.
+      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+      targetShell = window != null ? window.getShell() : getParentShell();
+    } catch (IllegalStateException ise) {  // getWorkbench() might throw this.
+      // No special handling. targetShell will be null.
+    }
 
-    int heightCaptionAndUpperBorder =
-        parentBounds.height - parentClientArea.height - getParentShell().getBorderWidth();
-    return new Point(parentBounds.x + parentClientArea.width - initialSize.x,
-        parentBounds.y + heightCaptionAndUpperBorder);
+    if (targetShell == null) {
+      return super.getInitialLocation(initialSize);
```

Comment:
Hmm, I could push the code into the try-catch block, but I think we still need the if-else. You don't want something like below, do you?

```java
  @Override
  protected Point getInitialLocation(Point initialSize) {
    try {
      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
      if (window != null) {
        return getTopRightLocation(window.getShell(), initialSize);
      }
      if (getParentShell() != null) {
        return getTopRightLocation(getParentShell(), initialSize);
      }
      return super.getInitialLocation(initialSize);
    } catch (IllegalStateException ise) {  // getWorkbench() might throw this.
      if (getParentShell() != null) {
        return getTopRightLocation(getParentShell(), initialSize);
      }
      return super.getInitialLocation(initialSize);
    }
  }

  private Point getTopRightLocation(Shell targetShell, Point initialSize) {
    // Position the dialog at the top-right corner of the targetShell window.
    Rectangle parentBounds = targetShell.getBounds();
    Rectangle parentClientArea = targetShell.getClientArea();

    int heightCaptionAndUpperBorder =
        parentBounds.height - parentClientArea.height - targetShell.getBorderWidth();
    return new Point(parentBounds.x + parentClientArea.width - initialSize.x,
        parentBounds.y + heightCaptionAndUpperBorder);
  }
```